kernel/log.h

```diff
@@ -154,7 +153,7 @@ static inline void log_assert_worker(bool cond, const char *expr, const char *fi
 }
 #  define log_assert(_assert_expr_) YOSYS_NAMESPACE_PREFIX log_assert_worker(_assert_expr_, #_assert_expr_, __FILE__, __LINE__)
 #else
-#  define log_assert(_assert_expr_)
+#  define log_assert(_assert_expr_) (void)(_assert_expr_);
```

Comment:
If `_assert_expr_` has side effects, or the compiler can't tell that it has none, it is still being executed. We explicitly do not want that, as `_assert_expr_` may do expensive things like looking up IdStrings. Better would be for example `do { if (0) ... }`.

Reply:
Why was this marked "resolved"? I haven't seen this issue addressed.

Reply:
Oops sorry. I forgot to push.

Reply:
Ah.  :+1: